path.go

```diff
@@ -116,7 +139,10 @@ func CleanPath(p string) string {
 	if len(buf) == 0 {
 		return p[:w]
 	}
-	return string(buf[:w])
+
+	cPath := string(buf[:w])
+	putBack(buf)
```

Comment:
you're also putting back the supposed stack buffer. Thus, it is marked as "escaping" and actually allocated on the heap

Reply:
That is the point, it seems that no way to avoid allocations on the heap when using sync.Pool, so I guess we are abandoning sync.Pool?

Reply:
I think you could avoid it by something like that:
```go
var poolBuf []byte

...

poolBuf = getLen(n + 1)
buf = poolBuf

...

if poolBuf != nil {
    putBack(poolBuf)
}
```

Most likely https://github.com/julienschmidt/httprouter/commit/92dde9c3b4783f3ca4e5c84a692531b623b4d49a is the more efficient approach. But you're also almost done, so let's benchmark them both.

Reply:
I don't that is going to work cuz the root cause of "escaping" variable `buf` is from `sync.Pool.Put(interface)` not `putBuf`, so sync.Pool will always make allocs on the heap.